### PR TITLE
fix: image-upload

### PR DESCRIPTION
**Pull request by SalmanMughal789**
This pull request addresses the previously reported issue with image uploads for order items and invoice items.

 Users can now successfully upload images for both entities within the system.

Fixes:

    Resolved bug preventing image uploads for order items.
    Addressed issue causing failed uploads for invoice items.
    
It also addresses the editable functionality for items and for order-items. The state data point is also managed in here.

The use of standard paylaod is used. The image selector works fine. It also accurately adds rows after uploading inside an order or in invoice.

### CODE REVIEW

**Comment by Waqar-SE** [comment]
This branch should have been created from `order-issue` branch. Now it can't be add-item branch can't be merged as long as we don't merge this one first. Please keep a track of such things.

File: frontend/context/BillContext.jsx [hunk 1]
```diff
@@ -151,20 +150,23 @@ export function BillProvider({ children }) {
         break
 
       case "addRow":
-        // const item = addItem(state, action);
-        if (action.payload.model === "order") {
+        // if payload has an model of order then this will mean order is to be added in invoice
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:153]: Restraint from long lines of comments, summarize them.

File: frontend/context/BillContext.jsx [hunk 2]
```diff
@@ -151,20 +150,23 @@ export function BillProvider({ children }) {
...
+        if (!action.payload.orderId && action.payload.model === "order") {
           newState.selectedInvoice.orders = [
             ...state.selectedInvoice.orders,
            addOrder(state)
           ]
         }
+        // If no orderId is found and model is item it would mean the item should be added
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:160]: Summarize.

File: frontend/context/BillContext.jsx [hunk 3]
```diff
@@ -176,38 +178,36 @@ export function BillProvider({ children }) {
...
         }
-        else {
-          newState.selectedInvoice.items = state.selectedInvoice?.items.map(item =>
-            item.id === action.payload.id ? updateRow(item, action.payload) : item
-          )
-        }
+
         break;
 
       case "setImageData":
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:196]: Hasn't this action become same as Setting width, heigh, price and qunantity. If yes then let's the above code by adding this case there.

File: frontend/context/BillContext.jsx [hunk 4]
```diff
@@ -176,38 +178,36 @@ export function BillProvider({ children }) {
...
-        let orderedIndex = getOrderIndex(state, action.payload.objectId);
-        if (typeof orderedIndex !== "undefined" && orderedIndex !== -1) {
+        debugger;
+        if(!action.payload.orderId && action.payload.itemId) {
+            newState.selectedInvoice.items = state.selectedInvoice?.items.map(item =>
+            item.id === action.payload.itemId ? updateRow(item, action.payload) : item )
+        }
+        else{
+
+          let orderedIndex = getOrderIndex(state, action.payload.orderId);
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:188]: use const for variable instead.

**Review by Waqar-SE — CHANGES_REQUESTED** [review]
Looks good but needs minor changes.

**Comment by SalmanMughal789** [comment]
> Looks good but needs minor changes.



> This branch should have been created from `order-issue` branch. Now it can't be add-item branch can't be merged as long as we don't merge this one first. Please keep a track of such things.

Actually I had to do this work on top of add items fix because If I cannot add items how could I test whether the image is uploaded correctly or not. This is why I needed to checkout to this branch from add item fix branch.

**Pull request merged**